examples/net_server/main.rs

```diff
@@ -35,31 +35,53 @@ impl SimpleState for State1 {
     fn on_start(&mut self, _data: StateData<'_, GameData<'_, '_>>) {}
 }
 
-/// A simple system that receives a ton of network events.
-struct SpamReceiveSystem {
-    pub reader: Option<ReaderId<NetEvent<String>>>,
+/// Component to store client's event subscription
+struct SpamReader {
+    reader: ReaderId<NetEvent<String>>,
+}
+
+impl Component for SpamReader {
+    type Storage = VecStorage<Self>;
 }
 
+/// A simple system that receives a ton of network events.
+struct SpamReceiveSystem {}
+
 impl SpamReceiveSystem {
     pub fn new() -> Self {
-        SpamReceiveSystem { reader: None }
+        SpamReceiveSystem {}
     }
 }
 
 impl<'a> System<'a> for SpamReceiveSystem {
-    type SystemData = (WriteStorage<'a, NetConnection<String>>, Entities<'a>);
-    fn run(&mut self, (mut connections, entities): Self::SystemData) {
+    type SystemData = (
+        WriteStorage<'a, NetConnection<String>>,
+        WriteStorage<'a, SpamReader>,
+        Entities<'a>,
+    );
+    fn run(&mut self, (mut connections, mut readers, entities): Self::SystemData) {
         let mut count = 0;
         let mut connection_count = 0;
 
         for (e, connection) in (&entities, &mut connections).join() {
-            if self.reader.is_none() {
-                self.reader = Some(connection.register_reader());
-            }
+            let spam_reader = readers.get_mut(e);
```

Comment:
This could make use of the Entry API.

Reply:
Thanks. I didn't find it myself. Fixed it.